src/pages/Causes/SignupForm.vue

```diff
@@ -0,0 +1,195 @@
+									v-for="(option, index) in mgAmountOptions"
+									:value="option.value"
+									:key="index"
+								>
+									{{ option.label }}
+								</option>
+							</kv-select>
+
+							<p class="tw-mt-2 tw-text-base">
+								<!-- eslint-disable-next-line max-len -->
+								Your subscription will renew automatically every month on the 1st. You can cancel any time.
+							</p>
+
+							<causes-drop-in-payment-wrapper
+								class="tw-mt-4 tw-mx-auto"
+								:amount="amount"
+								:cause-category-id="causeCategoryId"
+								@complete-transaction="completeCausesBraintree"
+							/>
+						</fieldset>
+					</form>
+				</div>
+			</kv-grid>
+		</kv-page-container>
+	</section>
+</template>
+
+<script>
+import gql from 'graphql-tag';
+
+import { validationMixin } from 'vuelidate';
+import numeral from 'numeral';
+import { required } from 'vuelidate/lib/validators';
+
+import CausesDropInPaymentWrapper from '@/components/Causes/CausesDropInPaymentWrapper';
+import KvGrid from '~/@kiva/kv-components/vue/KvGrid';
+import KvPageContainer from '~/@kiva/kv-components/vue/KvPageContainer';
+import KvSelect from '~/@kiva/kv-components/vue/KvSelect';
+
+const causesIconImgRequire = require.context('@/assets/images/causes-icons/', true);
+
+const pageQuery = gql`query causesCategoryIds {
+  getCategories(subscriptionType: "CAUSES") {
+    totalCount
+    values {
+      enabled
+      id
+      name
+    }
+  }
+}`;
+
+export default {
+	props: {
+		cause: {
+			type: String,
+			default: 'climate',
+			validator: value => {
+				return ['climate', 'education', 'women'].indexOf(value) !== -1;
+			}
+		},
+	},
+	components: {
+		CausesDropInPaymentWrapper,
+		KvGrid,
+		KvPageContainer,
+		KvSelect,
+	},
+	metaInfo: {
+		title: 'Get Started',
+	},
+	mixins: [validationMixin],
+	validations: {
+		amount: {
+			required,
+		},
+	},
+	data() {
+		return {
+			causesIconImgRequire,
+			mgAmountOptions: [
+				{
+					value: 5,
+					label: `${numeral(5).format('$0,0')}`,
+				},
+				{
+					value: 15,
+					label: `${numeral(15).format('$0,0')}`,
+				},
+				{
+					value: 25,
+					label: `${numeral(25).format('$0,0')}`,
+				},
+				{
+					value: 50,
+					label: `${numeral(50).format('$0,0')}`,
+				},
+				{
+					value: 75,
+					label: `${numeral(75).format('$0,0')}`,
+				},
+				{
+					value: 100,
+					label: `${numeral(100).format('$0,0')}`,
+				},
+			],
+			amount: 5,
+			success: false,
+			causesCategories: [],
+		};
+	},
+	inject: ['apollo', 'cookieStore'],
+	apollo: {
+		query: pageQuery,
+		preFetch: true,
+		result(result) {
+			this.causesCategories = result?.data?.getCategories?.values ?? [];
+		},
+	},
+	methods: {
+		completeCausesBraintree() {
+			// disable form inputs while routing to thanks
+			this.success = true;
+			this.$kvTrackEvent('Registration', 'successful-causes-reg', 'register-causes');
+			this.$router.push({
+				path: 'causes/thanks',
+			});
+		},
+	},
+	computed: {
+		causeCategoryId() {
```

Comment:
This feels super hacky. But we need a way to get the category id from the api for a specific category. Obviously if the name changes this would break. And it would be nice to use the name as the display name on the ui as well. 

I think a better approach is to maybe add a key to the category that will never change and we can use that to map key to category id. @mcstover thoughts on this?

Reply:
I'd agree that it would be better to have an ID passed around to reference a specific "cause" from your list. That may be a future piece of work though as I think you'd need to be passing the id from the initial Cause selection through login and to this page.